tests/pos/decorators/DocComment.scala

```diff
@@ -0,0 +1,26 @@
+/** Represents a doc comment, splitting it into `body` and `tags`
+ *  `tags` are all lines starting with an `@`, where the tag thats starts
+ *         with `@` is paired with the text that follows, up to the next
+ *         tagged line.
+ *  `body` what comes before the first tagged line
+ */
+case class DocComment(body: String, tags: Map[String, List[String]])
```

Comment:
Note that we already have logic for parsing documentation comments in the compiler for the benefit of dottydoc: https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/util/CommentParsing.scala / https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/Comments.scala

Reply:
Yes. The file here was meant as a demonstrator that it's easy to parse doc comments. Maybe we should make `CommentParsing.scala` generally available instead. It will be work to document everything properly though since it's a large API.